src/DynamoCoreWpf/UI/HeaderTemplateSelector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using Dynamo.UI.Controls;
+
+namespace Dynamo.Controls
+{
+    public class HeaderTemplateSelector : DataTemplateSelector
```

Comment:
4. Border has round corners for multiple items.

As I see what are you proposes can be done in one or many additional converters, triggers added to existing listbox item style. It looks like we move `TemplateSelector` complexity to `Style`. For some new difference we will need to add new trigger, converter. 

Other option is to specify `ItemContainerStyleSelector` and use two different styles. I like having two separate styles. Each of them will specify only that properties they need. However `ItemContainerStyle` way is generally the same as current implementation with `TemplateSelector`.

@Benglin, which way do you prefer?